sklearn/naive_bayes.py

```diff
@@ -117,6 +118,320 @@ def predict_proba(self, X):
+            raise NotFittedError("Call the fit() method first "
+                                 "before calling predict().")
+
+        # Obtain the log priors from each fitted estimator
+        all_log_priors = [
+            nb_model.class_log_prior_
+            if hasattr(nb_model, 'class_log_prior_') else np.log(nb_model.class_prior_)
+            for _, nb_model, _ in self.models_]
+
+        # Ensure class log priors are the same for all estimators
+        all_log_priors = np.hstack([all_log_priors])
+        if np.max(np.ptp(all_log_priors, axis=0)) < 1e-6:
+            log_prior = all_log_priors[0]
+        else:
+            raise ValueError("Class priors for every estimator "
+                             "must be the same.")
+
+        # Obtain the jll of each fitted estimator
+        jlls = [nb_model._joint_log_likelihood(
+                    np.array(_safe_indexing(X, cols, axis=1)))
+                for (_, nb_model, cols) in self.models_]
+
+        # Stack these jlls to give us
+        # the shape (estimator, sample, class)
+        jlls = np.hstack([jlls])
+
+        # Subtract the class log prior from all the jlls
+        # but add it back after the summation
+        jlls = jlls - log_prior
+        jll = jlls.sum(axis=0) + log_prior
+
+        return jll
+
+    @property
+    def _models(self):
+        return [(name, model) for name, model, _ in self.models]
+
+    @_models.setter
+    def _models(self, value):
+        self.models = [
+            (name, nb_model, cols) for ((name, nb_model), (_, _, cols))
+            in zip(value, self.models)]
+
+    def get_params(self, deep=True):
+        return self._get_params('_models', deep=deep)
+
+    def set_params(self, **kwargs):
+        self._set_params('_models', **kwargs)
+        return self
+
+    def _validate_models(self, X):
+
+        valid_modules = copy.copy(__all__)
+        valid_modules.remove("GeneralNB")
+
+        self._cols = []
+        dict_col2model = {}
+        all_fit_priors = []
+        all_class_priors = []
+
+        # Check type of `models` parameter
+        if not isinstance(self.models, list):
+            raise TypeError(
+                "Expected list but got {}".format(type(self.models)))
+
+        names, _, _ = zip(*self.models)
+        self._validate_names(names)
+
+        for model in self.models:
+
+            # Check type of each entry in list
+            if not isinstance(model, tuple):
+                raise TypeError(
+                    "Expected list of tuples "
+                    "but got list of {}s".format(type(model)))
+            if len(model) != 3:
+                raise ValueError("Expected tuple to have length of 3 "
+                                 "but got {}".format(len(model)))
+
+            _, estimator, cols = model
+
+            # Check naive bayes estimator for format
+            # `fit` and `_joint_log_likelihood` attributes
+            if callable(estimator):
```

Comment:
This is to check if the users specify `GeneralNB([("gaussian", GaussianNB, [0, 1, 2])])` instead of `GeneralNB([("gaussian", GaussianNB(), [0, 1, 2])])` (with and without parentheses). Is this an unnecessary check?

Reply:
It's a reasonable check, I suppose but hard for the reader to understand without a comment

Reply:
Alright, will add some comments.